### PR TITLE
Add option to get package.json content

### PR DESCRIPTION
**Pull request by helfi92**
References https://github.com/mozilla-neutrino/neutrino-dev/issues/516.

### CODE REVIEW

File: packages/neutrino/src/api.js [hunk 1]
```diff
@@ -49,6 +50,8 @@ class Api {
       });
     });
 
+    options.packageJson = require(path.join(options.root, 'package.json')); // eslint-disable-line global-require
```
Comment (eliperelman) [inline packages/neutrino/src/api.js]: The section you removed from the node preset is essentially what we need here. We need to try to require the file, but not blow up if it's not there. Maybe we should consider setting this to an empty object? Not sure...

Comment (eliperelman) [reply packages/neutrino/src/api.js]: Meh, we can leave it undefined.

**Comment by eliperelman** [comment]
Any thoughts on the option name? While `packageJson` works, it seems kinda awkward. cc: @edmorley

**Comment by helfi92** [comment]
I hear you, I think it looks awkward too. `pkgJson` is slightly better. I don't like `package` just because it's hard to know what kind of package it is by simply looking at the name.

**Comment by eliperelman** [comment]
I'm just annoyed at the `Json` part. I think I prefer `package` over `packageJson`, but neither make me happy. Maybe there's another word we can use?

**Comment by eliperelman** [comment]
Meh, if anything we should probably roll with `packageJson` since it's the most descriptive and relevant. Let's see what @edmorley thinks.

**Comment by edmorley** [comment]
`packageMetadata`, `projectMetadata`, ...?

I can't think of a great name either really! :-)

**Comment by helfi92** [comment]
OK, reverting back to `packageJson` 🤢 hehe

File: packages/neutrino/src/api.js [hunk 2]
```diff
@@ -49,6 +50,12 @@ class Api {
       });
     });
 
+    try {
+      options.packageJson = require(path.join(options.root, 'package.json')); // eslint-disable-line global-require
+    } catch (err) {
+      options.packageJson = undefined;
```
Comment (eliperelman) [inline packages/neutrino/src/api.js]: Let's use `null` here instead.

File: packages/neutrino/src/api.js [hunk 3]
```diff
@@ -2,6 +2,7 @@ const Config = require('webpack-chain');
 const merge = require('deepmerge');
 const Future = require('fluture');
 const mitt = require('mitt');
+const path = require('path');
```
Comment (eliperelman) [inline packages/neutrino/src/api.js]: Destructure out `join`.

File: packages/library/index.js [hunk 4]
```diff
@@ -64,7 +64,8 @@ module.exports = (neutrino, opts = {}) => {
   }
 
   try {
-    const pkg = require(join(neutrino.options.root, 'package.json')); // eslint-disable-line global-require
+    const pkg = neutrino.options.packageJson;
+
```
Comment (eliperelman) [inline packages/library/index.js]: Remove extra blank line.

**Comment by eliperelman** [comment]
Also need to add blurbs to the docs about this option.

**Comment by eliperelman** [comment]
We can do that in another bug if you like.

**Comment by helfi92** [comment]
Yes, will open up a different PR for this. Thanks for the reminder.

**Pull request merged**